From 6a2bd92f239f1818243af45685f12084554e35bb Mon Sep 17 00:00:00 2001
From: ltian <le.tian@windriver.com>
Date: Fri, 12 Nov 2010 15:54:20 +0800
Subject: [PATCH 048/132] Cavium: Provide definition of arch_randomize_brk

Source: SDK 2.0.0-366

32-bit processes have the heap randomized in an 8MB space, 256MB for
64-bit processes.

Signed-off-by: ltian <le.tian@windriver.com>
---
 arch/mips/include/asm/elf.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/mips/include/asm/elf.h b/arch/mips/include/asm/elf.h
index ea77a42..fd1d39e 100644
--- a/arch/mips/include/asm/elf.h
+++ b/arch/mips/include/asm/elf.h
@@ -372,4 +372,9 @@ extern const char *__elf_platform;
 struct linux_binprm;
 extern int arch_setup_additional_pages(struct linux_binprm *bprm,
 				       int uses_interp);
+
+struct mm_struct;
+extern unsigned long arch_randomize_brk(struct mm_struct *mm);
+#define arch_randomize_brk arch_randomize_brk
+
 #endif /* _ASM_ELF_H */
-- 
1.6.5.2

